src/palace/manager/service/sitewide.py

```diff
@@ -1,28 +1,24 @@
 from __future__ import annotations
 
-from typing import Literal, cast
+from typing import Literal
+from urllib.parse import urlparse
 
-from pydantic import AnyHttpUrl, NonNegativeInt, validator
+from pydantic import NonNegativeInt, field_validator
 
 from palace.manager.service.configuration.service_configuration import (
     ServiceConfiguration,
 )
+from palace.manager.util.pydantic import HttpUrl
 
 
 class SitewideConfiguration(ServiceConfiguration):
-    base_url: AnyHttpUrl | None = None
-    patron_web_hostnames: list[AnyHttpUrl] | Literal["*"] = []
+    base_url: HttpUrl | None = None
+    patron_web_hostnames: list[HttpUrl] | Literal["*"] = []
     authentication_document_cache_time: NonNegativeInt = 3600
     quicksight_authorized_arns: dict[str, list[str]] | None = None
 
-    @validator("base_url")
-    def validate_base_url(cls, v: AnyHttpUrl | None) -> AnyHttpUrl | None:
-        # Our base url should not end with a slash, if it does we remove it.
-        if v is not None and v.endswith("/"):
-            return cast(AnyHttpUrl, v.rstrip("/"))
-        return v
```

Comment:
The new pydantic URL validation does this automatically, so we just drop this validator.